### PR TITLE
Allow HIP and SYCL in space-time-stack

### PR DESCRIPTION
**Pull request by masterleinad**
With this patch information for allocations in `Host`, `Cuda`, `HIP`, and `SYCL` memory spaces are recorded in `space-time-stack`.

### CODE REVIEW

File: profiling/space-time-stack/kp_space_time_stack.cpp [hunk 1]
```diff
@@ -76,16 +77,26 @@ struct SpaceHandle {
...
-enum { NSPACES = 2 };
+enum { NSPACES = 4 };
 
 Space get_space(SpaceHandle const& handle) {
-  switch (handle.name[0]) {
-    case 'H': return SPACE_HOST;
-    case 'C': return SPACE_CUDA;
-  }
+  if (strcmp(handle.name, "Cuda") == 0 || strcmp(handle.name, "CudaUVM") == 0 ||
+      strcmp(handle.name, "CudaHostPinned")==0 )
```
Comment (dalg24) [inline profiling/space-time-stack/kp_space_time_stack.cpp:89]:
```suggestion
  if (strncmp(handle.name, "Cuda", 4) == 0)  // check that name starts with "Cuda"
```

**Review by dalg24 — APPROVED** [review]
I would use `strncmp` but not blocking

**Review by DavidPoliakoff — CHANGES_REQUESTED** [review]
Agree with Damien's comment, but I'll block on it. This way if we do something weird to the Cuda backend, we still should see this work

File: profiling/space-time-stack/kp_space_time_stack.cpp (hunk 1 above)
Comment (masterleinad) [reply profiling/space-time-stack/kp_space_time_stack.cpp]: Changed for `CUDA`, `HIP` and `SYCL`.

**Comment by masterleinad** [comment]
@DavidPoliakoff Is there anything else you want me to do here?

**Review by DavidPoliakoff — APPROVED** [review]

**Comment by DavidPoliakoff** [comment]
@masterleinad: in the long term, we're phasing out develop, but for the moment could you retarget against develop?

**Comment by masterleinad** [comment]
Sure, targeting `develop` now.

**Comment by DavidPoliakoff** [comment]
I don't know how I forgot to merge this, apologies

**Pull request merged**